### PR TITLE
Mats and Evelina's labyrinth

### PR DESCRIPTION
**Pull request by evelinabe**
Link to my deployed project: https://labyrinth-2020.netlify.app/

Our reflections on how this week's project turned out: We had some issues with fetching the games data from the API and in the end we noticed that the problem was due to a naming error in the function that was executing the fetch.

### CODE REVIEW

File: code/src/components/Loader.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      animationData: animationData,
+      rendererSettings: {
+        preserveAspectRatio: "xMidYMid slice"
+      }
+    };
+  
+  return (
+    <LoaderContainer>
+      <Lottie 
+	      options={defaultOptions}
```
Comment (emma-tech) [inline code/src/components/Loader.js:20]: To much indent here! :)

File: code/src/components/Loader.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import { LoaderContainer } from '../lib/LoaderStyle';
+
+
+export const Loader = () => {
+  const defaultOptions = {
+      loop: true,
+      autoplay: true,
+      animationData: animationData,
+      rendererSettings: {
+        preserveAspectRatio: "xMidYMid slice"
```
Comment (emma-tech) [inline code/src/components/Loader.js:13]: To much indent here! :)

File: code/src/reducers/game.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            state.gameDetails = action.payload;
+            console.log(state.gameDetails)
+         },
+         restartGame: (state, action) => {
+             return initialState
+         },
+         setLoader: (state, action) => {
+            state.isLoading = action.payload;
+         },
+        // setHistory: (state, action) => {
```
Comment (emma-tech) [inline code/src/reducers/game.js:33]: Don't forget to remove out-commented code before last commit! :)

File: code/src/reducers/reusable.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({
+                username: userName,
+            }),
+        })
+            .then(res => res.json())
+            .then(data => {
+                console.log(data);
+ //               localStorage.setItem("game", JSON.stringify(json[0]));
```
Comment (emma-tech) [inline code/src/reducers/reusable.js:17]: Some out-commented code here :)

File: code/src/reducers/reusable.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { game } from './game';
+
+
+export const startFetch = (userName) => {
+    console.log("hellog")
```
Comment (emma-tech) [inline code/src/reducers/reusable.js:5]: Don't forget to remove console.log()'s  before last commit :)

File: code/src/components/GameContinue.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import { Container, Section, Text, Description, Wrapper, GameButtonWrapper } from '../lib/GameContinueStyle';
+import { GameButton } from '../lib/Button';
+
+export const GameContinue = () => {
+  const dispatch = useDispatch();
+  const userName = useSelector((store) => store.game.userName);
+  const games = useSelector((store) => store.game.gameDetails);
+  const loader = useSelector(store => store.game.isLoading);
+
+
```
Comment (emma-tech) [inline code/src/components/GameContinue.js:15]: Some extra whitespace here that can be removed :)

File: code/src/components/GameStart.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { game } from '../reducers/game';
+import { startFetch } from '../reducers/reusable';
+import { GameHeader } from '../lib/GameContainerStyle';
+import { GameButton } from '../lib/Button';
+import { InputForm } from '../lib/GameStartStyle';
+
+export const GameStart = () => {
```
Comment (emma-tech) [inline code/src/components/GameStart.js:9]: Your code is so well structured and clean, really easy to read 🌟👏

File: code/src/components/GameStart.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+        <InputForm>
+            <label
+                value='username'
+            />
+            <input
+                type='text'
+                placeholder='Type username'
+                onChange={e => setUserNameInput(e.target.value)}
+            />
+            <GameButton onClick={handleSubmit} width={100} disabled={!userNameInput}>Let's go!</GameButton>
```
Comment (emma-tech) [inline code/src/components/GameStart.js:35]: Really good that you disabled the button if the user hasn't typed in a user name (will change that in our own application as well! 👍)

File: code/src/components/Loader.js [hunk 9]
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import Lottie from 'react-lottie';
+import animationData from '../lib/lotties/40235-si-fi-circle.json';
```
Comment (emma-tech) [inline code/src/components/Loader.js:3]: Really cool loader 😍

File: code/src/lib/Button.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import styled from 'styled-components';
```
Comment (emma-tech) [inline code/src/lib/Button.js:1]: Really good that you're so consistent with your styling, that you use styled components and that you have structured them using these lib/ folders 🌟

File: code/src/reducers/game.js [hunk 11]
```diff
@@ -0,0 +1,38 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+// const initialContent = localStorage.getItem('game')
```
Comment (emma-tech) [inline code/src/reducers/game.js:3]: Don't forget to remove out-commented code :)

**Review by emma-tech — COMMENTED** [review]
Your project looks really good Mats and Evelina! The code is clean, well structured and easy to read. We're really impressed with how you've structured your styling as well, will be some inspiration for us to take with us to future projects 👍

We have left some comments on out-commented code and extra whitespace that we found in the project, so maybe have a look at that :) Otherwise you've met all the requirements and we thinks you project is approved 🌟👏

//Emma, Petra, Sara

**Comment by emma-tech** [comment]
You've reached the requrements for this week and we think the project is approved 👍

**Review by emma-tech — APPROVED** [review]

File: code/src/components/GameContainer.js [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+import { GameStart } from './GameStart';
+
+
+const GameContainer = () => {
+    const gameStart = useSelector((store) => store.game.isGameStarted); 
+
+    return (
+        <StartGameContainer>
+            <GameStart />
+            {gameStart ? <GameContinue /> : <></>}
```
Comment (puj) [inline code/src/components/GameContainer.js:14]: `{gameStart  && <GameContinue /> }` works the same as this

**Review by puj — CHANGES_REQUESTED** [review]
Nice job.  Good result and clean, logical code along the way.  Changes requested before approval:
- Remove commented code
- Remove `console.log`s

**Comment by evelinabe** [comment]
> Nice job. Good result and clean, logical code along the way. Changes requested before approval:
> 
> * Remove commented code
> * Remove `console.log`s

Fixed!

**Pull request closed without merging**